### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'chains')

### ISSUE THREAD

**Issue by gsw85** [rainbow-me/rainbowkit#1225]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### RainbowKit Version

1.0.0

### wagmi Version

1.0.1

### Current Behavior

It was ok before 1.0.0 version

Getting this error:

- error node_modules\@rainbow-me\rainbowkit\dist\chunk-BZODBG62.js (629:0) @ useMainnet
- error TypeError: Cannot read properties of undefined (reading 'chains')

### Expected Behavior

Loading the connect button regulary

### Steps To Reproduce

```
import { ConnectButton } from "@rainbow-me/rainbowkit";
import "@rainbow-me/rainbowkit/styles.css";
import {
  getDefaultWallets,
  RainbowKitProvider,
  lightTheme,
} from "@rainbow-me/rainbowkit";
import { configureChains, createConfig, WagmiConfig } from "wagmi";
import { mainnet, polygon, optimism, arbitrum } from "wagmi/chains";
import { alchemyProvider } from "wagmi/providers/alchemy";
import { publicProvider } from "wagmi/providers/public";

export function WagmiAuth({ children }) {
  const myCustomTheme = merge(lightTheme(), {
    colors: { accentColor: "rgb(202 138 4)" },
    fonts: {
      body: "-apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif",
    },
  });

  const { chains, provider } = configureChains(
    [mainnet, polygon, optimism, arbitrum],
    [
      alchemyProvider({ apiKey: process.env.NEXT_PUBLIC_ALCHEMY_PROVIDER }),
      publicProvider(),
    ]
  );

  const { connectors } = getDefaultWallets({
    appName: "Company",
    learnMoreUrl: "https://www.company.com",
    projectId: process.env.NEXT_PUBLIC_WALLETCONNECT_PROJECT_ID,
    chains,
  });

  const wagmiConfig = createConfig({
    autoConnect: true,
    connectors,
    provider,
  });

  console.log(chains);

  return (
    <>
      <WagmiConfig config={wagmiConfig}>
        <RainbowKitProvider chains={chains} theme={myCustomTheme}>
          {children}
        </RainbowKitProvider>
      </WagmiConfig>
    </>
  );
}

```

**Comment by akshaynarisetti** [comment]
Facing the same issue, any solution here?

**Comment by cxss** [comment]
Facing the same issue with "viem": "^0.3.18",     "wagmi": "^1.0.0",  "@rainbow-me/rainbowkit": "^1.0.0",

**Comment by DanielSinclair** [comment]
What are the reproduction steps? Can't seem to replicate this. A reproducible example, or additional information about the frameworks and build system used would be helpful as well.

**Comment by akshaynarisetti** [comment]
```
    "@rainbow-me/rainbowkit": "^1.0.0",
    "viem": "~0.3.19",
    "wagmi": "^1.0.1"
``` 

The above dependencies fixed the issue for me. I picked them up from nextjs example from rainbowkit 🌈 directly

**Comment by DanielSinclair** [comment]
@gsw85 Do the above peer versions solve this issue for you?

**Comment by gsw85** [comment]
I am still getting tthe error of:
```
- error node_modules\@rainbow-me\rainbowkit\dist\chunk-BZODBG62.js (629:0) @ useMainnet
- error TypeError: Cannot read properties of undefined (reading 'chains')
null
```
This is the stack I am using:
"next": "^13.4.1",
"react": "^18.2.0",
"react-dom": "^18.2.0",

I am using Nextjs's app directory instead of pages, the following version was working:
"@rainbow-me/rainbowkit": "^0.12.12",
"wagmi": "^0.12.13",

and after updating it to the following, I am getting the above mentioned error:
"@rainbow-me/rainbowkit": "^1.0.0",
"viem": "~0.3.19",
"wagmi": "^1.0.1",

**Comment by gsw85** [comment]
> ```
>     "@rainbow-me/rainbowkit": "^1.0.0",
>     "viem": "~0.3.19",
>     "wagmi": "^1.0.1"
> ```
> 
> The above dependencies fixed the issue for me. I picked them up from nextjs example from rainbowkit 🌈 directly

What's the Nextjs version you are using? are you using the app directory ?

**Comment by jeffishjeff** [comment]
I have the same problem with below stack and app directory:

  "dependencies": {
    "@headlessui/react": "^1.7.14",
    "@rainbow-me/rainbowkit": "^1.0.0",
    "autoprefixer": "10.4.14",
    "next": "13.4.1",
    "postcss": "8.4.23",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "tailwindcss": "3.3.2",
    "viem": "^0.3.21",
    "wagmi": "^1.0.2"
  }

The only other change I made was on webpack config to fix the cannot resolve 'fs' issue:

const nextConfig = {
  webpack: (config) => {
    config.resolve.fallback = { fs: false, net: false, tls: false, encoding: false };
    return config;
  },
};

**Comment by jxom** [comment]
The code provided in the OP uses wagmi v0 APIs. You'll need to [migrate to the wagmi v1 APIs](https://wagmi.sh/react/migration-guide#provider--signer-terminology) – specifically, change `provider` to `publicClient`.

**Comment by maximedgr** [comment]
@jxom perfect it works!
So, it seems that the official Rainbowkit documentation is outdated.  
Here is my modified and working code :  

```
import '@rainbow-me/rainbowkit/styles.css';
import {
  getDefaultWallets,
  RainbowKitProvider,
} from '@rainbow-me/rainbowkit';
import { configureChains, createConfig, WagmiConfig } from 'wagmi';
import { mainnet, polygon, optimism, arbitrum } from 'wagmi/chains';
import { publicProvider } from 'wagmi/providers/public';
import YourApp from './YourApp';

const { chains, publicClient } = configureChains(
  [mainnet, polygon, optimism, arbitrum],
  [
    publicProvider()
  ]
);

const { connectors } = getDefaultWallets({
  appName: 'My RainbowKit App',
  projectId: 'YOUR_PROJECT_ID',
  chains
});

const wagmiConfig = createConfig({
  autoConnect: true,
  connectors,
  publicClient
})
 
function App() {
  return (
    <WagmiConfig config={wagmiConfig}>
      <RainbowKitProvider chains={chains}>
        <YourApp />
      </RainbowKitProvider>
    </WagmiConfig>
  )
}

export default App;
```

And package.json :
```
"dependencies": {
    "@rainbow-me/rainbowkit": "^1.0.0",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@testing-library/user-event": "^13.5.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-scripts": "5.0.1",
    "viem": "^0.3.22",
    "wagmi": "^1.0.2",
    "web-vitals": "^2.1.4"
  }
 ```

**Comment by DanielSinclair** [comment]
@maximedgr Thanks for spotting, fixed the docs #1241

**Comment by Olanetsoft** [comment]
> const { chains, publicClient } = configureChains(
>   [mainnet, polygon, optimism, arbitrum],
>   [
>     publicProvider()
>   ]
> );

Works fine. Thank you 😀
